### ISSUE TITLE
Section titles with links are badly rendered

### ISSUE THREAD

**Issue by Lysxia** [haskell/haddock#1054]
For example the [`Data.Traversable`](https://hackage.haskell.org/package/base-4.12.0.0/docs/Data-Traversable.html) doc looks like this:

![sc-2019-04-22T06:02:47-04:00](https://user-images.githubusercontent.com/2515201/56495630-61e0c980-64c4-11e9-85f1-5f349ec21061.png)

Apparently, because section headers are links, they should not have "interactive content", which includes links. (HTML5 reference: https://www.w3.org/TR/html50/text-level-semantics.html#the-a-element)

Minimized example:

```
module A
  ( -- * The 'C' class
    C
  ) where

class C a where
```

**Comment by harpocrates** [comment]
Nice catch! The way browsers end up rendering this invalid HTML is interesting too.

I propose we work around this by disabling the section header links when some sort of link exists in the section header HTML.

**Comment by harpocrates** [comment]
Fixed in db3725fe9d1aaa424356f290654fb04213ef6c4e.
